administrator/components/com_fields/helpers/fields.php

```diff
@@ -161,6 +161,9 @@ public static function getFields ($context, $item = null, $prepareValue = false)
 				}
 
 				$field->rawvalue = $field->value;
+				
+				//Store item id on field instance for later use			
+				$field->item_id = $item->id;
```

Comment:
I guess we can do that since the triggers to those events are now only in the getFields method.